In [1]:
import requests
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
r = requests.get('https://collectio*******************rch?q=sunflowers')
sunflowers = r.json()

In [3]:
sunflowers

{'total': 87,
 'objectIDs': [436524,
  484935,
  437112,
  210191,
  431264,
  397949,
  656530,
  480725,
  486590,
  485308,
  375281,
  705155,
  11922,
  2032,
  343052,
  20141,
  2019,
  208554,
  347980,
  403496,
  360837,
  437115,
  207869,
  761604,
  400581,
  423400,
  707887,
  223828,
  682927,
  57922,
  649813,
  79739,
  193938,
  339347,
  208218,
  202228,
  437984,
  436534,
  704667,
  696949,
  437329,
  436252,
  436580,
  363282,
  207753,
  822570,
  816522,
  487043,
  437526,
  203893,
  367434,
  190739,
  423237,
  262423,
  839296,
  367817,
  13644,
  18362,
  820022,
  648617,
  761731,
  814626,
  892559,
  207967,
  626832,
  262430,
  262428,
  707697,
  205386,
  262378,
  426804,
  664519,
  633526,
  369073,
  747779,
  436121,
  436144,
  852414,
  427872,
  436530,
  436535,
  436529,
  724844,
  437980,
  834585,
  834580,
  834765]}

In [4]:
sunflower_ids = sunflowers['objectIDs']  # сохранили всё по ключу

In [5]:
search_ids = [436524, 2032, 20149, 16822570, 437329]  # записываем варианты id

In [7]:
[i for i in search_ids if i in sunflower_ids]

[436524, 2032, 437329]

#                                                        Задание 2
       Отлично! На предыдущем шаге мы получили результаты поиска. Следущий этап – собрать информацию об этих объектах.
       
 где OBJECT_ID – id объекта.

Выполните запрос для объекта 437980. Для проверки соотнесите полученные параметры с их значениями.

In [8]:
# взяли запрос из задачи, подставили id объекта
r_test = requests.get(f'https://collec*************************8980')  
obj_test = r_test.json()

In [10]:
obj_test['GalleryNumber']

'825'

In [11]:
obj_test['title']

'Cypresses'

In [13]:
obj_test['culture']

''

In [14]:
obj_test['objectName']

'Painting'

In [15]:
obj_test['objectID']

437980

In [16]:
obj_test['department']

'European Paintings'

In [19]:
obj_test['artistDisplayName']

'Vincent van Gogh'

In [20]:
# собираем датафрейм для результатов поиска 
object_df = pd.DataFrame()

In [27]:
# информация об объекте
r = requests.get(f'https://co***********************ct_id}')
obj_tmp = r.json()

In [25]:
# формируем строчку для объекта
# в кавычках как ключ указываем название колонки, а в качестве значения передаем значение из obj_tmp
obj_tmp_df = pd.DataFrame({'objectID' : obj_tmp['objectID'],
                           'title' : [obj_tmp['title']],
                           'artistDisplayName': [obj_tmp['artistDisplayName']],
                           'department' : [obj_tmp['department']], 
                           'objectBeginDate' : [obj_tmp['objectBeginDate']], 
                           'objectEndDate' : [obj_tmp['objectEndDate']], 
                           'period' : [obj_tmp['period']], 
                           'objectName' : [obj_tmp['objectName']], 
                           'culture': [obj_tmp['culture']]
                        })

# Полная версия кода:

In [26]:
# собираем датафрейм для результатов поиска 
object_df = pd.DataFrame()
for object_id in sunflower_ids:
  # информация об объекте
  r = requests.get(f'https://******************************ct_id}')
  obj_tmp = r.json()

  # формируем строчку для объекта
  try:
    obj_tmp_df = pd.DataFrame({'objectID' : obj_tmp['objectID'],
                              'title' : [obj_tmp['title']],
                              'artistDisplayName': [obj_tmp['artistDisplayName']],
                              'department' : [obj_tmp['department']], 
                              'objectBeginDate' : [obj_tmp['objectBeginDate']], 
                              'objectEndDate' : [obj_tmp['objectEndDate']], 
                              'period' : [obj_tmp['period']], 
                              'objectName' : [obj_tmp['objectName']], 
                              'culture': [obj_tmp['culture']]
                          })
    # приклеиваем
    object_df = pd.concat([object_df, obj_tmp_df])
  except KeyError:
      continue

In [28]:
object_df.query("culture == 'Japan'").shape[0]

1

In [29]:
r = requests.get('https://collec************************cat')
cats = r.json()

In [30]:
cat_ids = cats['objectIDs']  # сохранили id

In [31]:
# собираем датафрейм для результатов поиска 
object_df = pd.DataFrame()

for object_id in cat_ids[:1000]:  # берем только 1000 объектов, как указано в задании
    # информация об объекте
    r = requests.get(f'https://collec********************ct_id}')
    obj_tmp = r.json()

    # формируем строчку для объекта
    try:
        obj_tmp_df = pd.DataFrame({'objectID' : obj_tmp['objectID'],
                                 'title' : [obj_tmp['title']],
                                 'artistDisplayName': [obj_tmp['artistDisplayName']],
                                 'department' : [obj_tmp['department']], 
                                 'objectBeginDate' : [obj_tmp['objectBeginDate']], 
                                 'objectEndDate' : [obj_tmp['objectEndDate']], 
                                 'objectName' : [obj_tmp['objectName']], 
                                 'culture': [obj_tmp['culture']],
                                 'period': [obj_tmp['period']]
                            })
        # приклеиваем
        object_df = pd.concat([object_df, obj_tmp_df])
    except KeyError:
        continue
        

In [32]:
object_df.period.value_counts().head(5)

Edo period (1615–1868)          444
                                108
Qing dynasty (1644–1911)         62
Ming dynasty (1368–1644)         36
Muromachi period (1392–1573)     30
Name: period, dtype: int64